### PR TITLE
[WIP] Add rustls support

### PR DESCRIPTION
**Pull request by kinosang**

### CODE REVIEW

**Comment by ark0f** [comment]
Is work still in progress?

**Comment by ark0f** [comment]
Thank you for contribution! Is work still in progress?

**Comment by kinosang** [comment]
@ark0f yes. there's some problem with webpki. though I'm not familiar with rustls, I'm still working on that.

**Comment by kinosang** [comment]
@ark0f I've tried, but I always got `invalid certificate: UnknownIssuer` error, I cannot solve it.

File: Cargo.toml [hunk 1]
```diff
@@ -25,6 +25,11 @@ tokio = "0.1"
 hyper-tls = { version = "0.3", optional = true }
 native-tls = { version = "0.2", optional = true }
 
+# `rustls` feature
+tokio-rustls = { version = "0.9", optional=true }
```
Comment (ark0f) [inline Cargo.toml:29]: Add whitspaces between `=`.

File: src/lib.rs [hunk 2]
```diff
@@ -83,6 +88,14 @@ where
         let args = (self, TlsConnector::new()?);
         Ok(HttpsConnector::from(args))
     }
+
+    /// Create a new connector with rustls support
+    #[cfg(feature = "rustls")]
+    pub fn with_tls(self) -> Result<HttpsConnector<Self>, io::Error> {
+        let cfg = std::sync::Arc::new(tokio_rustls::rustls::ClientConfig::new());
```
Comment (ark0f) [inline src/lib.rs:95]: Add `use` at the top of the function instead of full path.

**Review by ark0f — CHANGES_REQUESTED** [review]

**Comment by kinosang** [comment]
@ark0f this pr is still WIP, I'll update asap.

**Comment by ark0f** [comment]
Done in #2

**Pull request closed without merging**